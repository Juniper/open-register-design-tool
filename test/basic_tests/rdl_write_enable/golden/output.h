//   Ordt 230719.01 autogenerated file 
//   Input: ./rdl_write_enable/test.rdl
//   Parms: ./rdl_write_enable/test.parms
//   Date: Thu Jul 20 13:47:02 EDT 2023
//

#ifndef __FOO_REGISTER_MAP__
#define __FOO_REGISTER_MAP__

#include <bits.h>

/* FOO_REGISTERS memory map */
enum FOO_REGS {
  INTR_REG = 0x0,
  LOG_REG = 0x4,
  BIGREG = 0x8,
  CHILDMAP_32B_ARRAY_REG_0 = 0x100,
  CHILDMAP_32B_ARRAY_REG_1 = 0x104,
  CHILDMAP_32B_ARRAY_REG_2 = 0x108,
  CHILDMAP_32B_ARRAY_REG_3 = 0x10c,
  CHILDMAP_32B_ARRAY_REG_4 = 0x110,
  CHILDMAP_32B_ARRAY_REG_5 = 0x114,
  CHILDMAP_32B_ARRAY_REG_6 = 0x118,
  CHILDMAP_32B_ARRAY_REG_7 = 0x11c
};

/* INTR_REG register fields */
#define INTR_REG_COUNT_FLD GENMASK(7, 0)
#define INTR_REG_INTR1 BIT(8)
#define INTR_REG_INTR2 BIT(9)

/* LOG_REG register fields */
#define LOG_REG_LOG_ERR GENMASK(7, 0)
#define LOG_REG_LOG_VALUE GENMASK(15, 8)

/* BIGREG register fields */
#define BIGREG_FIELD64 GENMASK(63, 0)

/* CHILDMAP_32B_ARRAY_REG_0 register fields */
#define CHILDMAP_32B_ARRAY_REG_0_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_1 register fields */
#define CHILDMAP_32B_ARRAY_REG_1_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_2 register fields */
#define CHILDMAP_32B_ARRAY_REG_2_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_3 register fields */
#define CHILDMAP_32B_ARRAY_REG_3_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_4 register fields */
#define CHILDMAP_32B_ARRAY_REG_4_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_5 register fields */
#define CHILDMAP_32B_ARRAY_REG_5_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_6 register fields */
#define CHILDMAP_32B_ARRAY_REG_6_FIELD32 GENMASK(31, 0)

/* CHILDMAP_32B_ARRAY_REG_7 register fields */
#define CHILDMAP_32B_ARRAY_REG_7_FIELD32 GENMASK(31, 0)

#endif
