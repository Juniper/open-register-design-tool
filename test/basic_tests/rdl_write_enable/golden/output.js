/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_write_enable/test.rdl
 *   Parms: ./rdl_write_enable/test.parms
 *   Date: Sun Nov 17 17:01:12 EST 2019
 */

register_set foo "Registers for foo" {

  address = 0x0;
  register_width = 32;
  register intr_reg "intr_reg register" {
    address = 0x0;
    nop[22];
    integer intr2[1] "intr2 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer intr1[1] "intr1 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
    integer count_fld[8] "count_fld field" param {
      access_mode = READ_WRITE;
      reset = 0x55;
    };
  };
  
  register log_reg "log_reg register" {
    address = 0x4;
    category = { STATE };
    nop[16];
    integer log_value[8] "log_value field" param {
      access_mode = READ_ONLY;
      reset = 0x0;
    };
    integer log_err[8] "log_err field" param {
      access_mode = READ_WRITE;
      reset = 0x4;
    };
  };
  
  register bigreg "bigreg register" {
    address = 0x8;
    register_width = 64;
    integer field64[64] "field64 field" param {
      access_mode = READ_WRITE;
      reset = 0x0;
    };
  };
  
  register_set childmap_32b "childmap_32b register_set" {
    address = 0x100;
    
    register array_reg "array_reg register" {
      address = 0x0;
      repeat = 8;
      integer field32[32] "field32 field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x20;
  };
  
  register_set_size = 0x120;
};

