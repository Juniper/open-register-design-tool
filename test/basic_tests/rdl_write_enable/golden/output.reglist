//   Ordt 190725.01 autogenerated file 
//   Input: ./rdl_write_enable/test.rdl
//   Parms: ./rdl_write_enable/test.parms
//   Date: Sun Nov 17 17:03:03 EST 2019
//

0x0              size: 32   fields: 3    bits: 10        intr_reg
0x4              size: 32   fields: 2    bits: 16        log_reg
0x8              size: 64   fields: 1    bits: 64        bigreg
0x100            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_0
0x104            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_1
0x108            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_2
0x10c            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_3
0x110            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_4
0x114            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_5
0x118            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_6
0x11c            size: 32   fields: 1    bits: 32        childmap_32b.array_reg_7
