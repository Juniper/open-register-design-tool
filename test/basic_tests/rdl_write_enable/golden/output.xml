<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 190725.01 autogenerated file 
       Input: ./rdl_write_enable/test.rdl
       Parms: ./rdl_write_enable/test.parms
       Date: Sun Nov 17 17:08:32 EST 2019
 -->

<map version="190725.01">
  <id>foo</id>
  <baseaddr>0</baseaddr>
  <shorttext>foo registers</shorttext>
  <reg>
    <id>intr_reg</id>
    <shorttext>intr_reg register</shorttext>
    <baseaddr>0x0</baseaddr>
    <width>32</width>
    <field>
      <id>count_fld</id>
      <shorttext>count_fld field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
        <hwset></hwset>
        <hwclr></hwclr>
        <swwel></swwel>
      </hwinfo>
      <reset>55</reset>
      <lowidx>0</lowidx>
      <width>8</width>
      <hwmod></hwmod>
      <counter>
        <incr>
          <incrvalue>0x1</incrvalue>
        </incr>
      </counter>
    </field>
    <field>
      <id>intr1</id>
      <shorttext>intr1 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>8</lowidx>
      <width>1</width>
      <hwmod></hwmod>
      <intr>
        <type>LEVEL</type>
        <stickytype>STICKYBIT</stickytype>
      </intr>
    </field>
    <field>
      <id>intr2</id>
      <shorttext>intr2 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>9</lowidx>
      <width>1</width>
      <hwmod></hwmod>
      <intr>
        <type>LEVEL</type>
        <stickytype>STICKYBIT</stickytype>
      </intr>
    </field>
    <access>RW</access>
  </reg>
  <reg>
    <id>log_reg</id>
    <shorttext>log_reg register</shorttext>
    <baseaddr>0x4</baseaddr>
    <width>32</width>
    <field>
      <id>log_err</id>
      <shorttext>log_err field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>w</hwaccess>
      </hwinfo>
      <reset>4</reset>
      <lowidx>0</lowidx>
      <width>8</width>
      <hwmod></hwmod>
    </field>
    <field>
      <id>log_value</id>
      <shorttext>log_value field</shorttext>
      <access>RO</access>
      <hwinfo>
        <hwaccess>w</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>8</lowidx>
      <width>8</width>
      <hwmod></hwmod>
    </field>
    <catcode>32</catcode>
    <access>RW</access>
  </reg>
  <reg>
    <id>bigreg</id>
    <shorttext>bigreg register</shorttext>
    <baseaddr>0x8</baseaddr>
    <width>64</width>
    <field>
      <id>field64</id>
      <shorttext>field64 field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>na</hwaccess>
      </hwinfo>
      <reset>0</reset>
      <lowidx>0</lowidx>
      <width>64</width>
    </field>
    <access>RW</access>
  </reg>
  <map>
    <id>childmap_32b</id>
    <shorttext>childmap_32b registers</shorttext>
    <baseaddr>0x100</baseaddr>
    <extroot>1</extroot>
    <reg>
      <id>array_reg</id>
      <parentpath>childmap_32b</parentpath>
      <shorttext>array_reg register</shorttext>
      <baseaddr>0x100</baseaddr>
      <width>32</width>
      <reps>8</reps>
      <stride>0x4</stride>
      <field>
        <id>field32</id>
        <shorttext>field32 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0</reset>
        <lowidx>0</lowidx>
        <width>32</width>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x11f</highaddr>
  </map>
</map>
